cl/search/documents.py

```diff
@@ -1070,3 +1081,578 @@ def prepare(self, instance):
+            )
+            return midnight_pt(datetime_object)
+
+        return midnight_pt(instance.docket.date_argued)
+
+    def prepare_dateReargued(self, instance):
+        if instance.docket.date_reargued is None:
+            return
+
+        if isinstance(instance.docket.date_reargued, str):
+            datetime_object = datetime.strptime(
+                instance.docket.date_reargued, "%Y-%m-%d"
+            )
+            return midnight_pt(datetime_object)
+
+        return midnight_pt(instance.docket.date_reargued)
+
+    def prepare_dateReargumentDenied(self, instance):
+        if instance.docket.date_reargument_denied is None:
+            return
+
+        if isinstance(instance.docket.date_reargument_denied, str):
+            datetime_object = datetime.strptime(
+                instance.docket.date_reargument_denied, "%Y-%m-%d"
+            )
+            return midnight_pt(datetime_object)
+
+        return midnight_pt(instance.docket.date_reargument_denied)
+
+    def prepare_neutralCite(self, instance):
+        neutral_citations = instance.citations.filter(type=Citation.NEUTRAL)
+        if neutral_citations.count():
+            return str(neutral_citations[0])
+        return ""
+
+    def prepare_lexisCite(self, instance):
+        lexis_citations = instance.citations.filter(type=Citation.LEXIS)
+        if lexis_citations.count():
+            return str(lexis_citations[0])
+        return ""
+
+    def prepare_timestamp(self, instance):
+        return datetime.utcnow()
+
+
+@opinion_index.document
+class OpinionDocument(OpinionBaseDocument):
+    id = fields.IntegerField(
+        attr="pk",
+        fields={
+            "raw": fields.KeywordField(attr="pk"),
+        },
+    )
+    author_id = fields.IntegerField()
+    type = fields.KeywordField(attr="type")
+    per_curiam = fields.BooleanField(attr="per_curiam")
+    type_text = fields.TextField(
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(analyzer="english_exact"),
+        },
+        search_analyzer="search_analyzer",
+    )
+    download_url = fields.KeywordField(attr="download_url", index=False)
+    local_path = fields.KeywordField(index=False)
+    text = fields.TextField(
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(analyzer="english_exact"),
+        },
+        search_analyzer="search_analyzer",
+    )
+    sha1 = fields.TextField(attr="sha1")
+    cites = fields.ListField(
+        fields.IntegerField(multi=True),
+    )
+    joined_by_ids = fields.ListField(
+        fields.IntegerField(multi=True),
+    )
+
+    class Django:
+        model = Opinion
+        ignore_signals = True
+
+    def prepare_author_id(self, instance):
+        return getattr(instance.author, "pk", None)
+
+    def prepare_type_text(self, instance):
+        return instance.get_type_display()
+
+    def prepare_local_path(self, instance):
```

Comment:
On RECAP we decided to only do something like:

```
    def prepare_filepath_local(self, instance):
        if instance.filepath_local:
            return instance.filepath_local.name
```

Since checking for the existence of the file before indexing will require millions of queries to S3, we simply index the path without verifying its availability on S3. I believe this approach applies only to Opinions, given its massive index size.

Reply:
understood!